plugins/statistics.koplugin/main.lua

```diff
@@ -1295,16 +1295,15 @@ function ReaderStatistics:getCurrentStat(id_book)
     local time_to_read = (self.data.pages - self.view.state.page) * self.avg_time
     local estimate_days_to_read = math.ceil(time_to_read/(book_read_time/tonumber(total_days)))
     local estimate_end_of_read_date = os.date("%Y-%m-%d", tonumber(now_ts + estimate_days_to_read * 86400))
-    local formatstr = "%.0f%%"
+    local estimates_valid = time_to_read > 0 -- above values could be 'nan' and 'nil'
```

Comment:
Which *above values* would that be?

Because if it's `time_to_read`, that won't work (can't compare nil to a number, at the very least. Not quite sure about NaN).

Reply:
> Which above values would that be?

Implicit, but `estimates_valid` is about the ones starting with `estimate_` .

I had no crash, I was just seeing NaN and nil for these 2 estimates on a new book without any page_stat_data entry yet.

Reply:
Oh, okay, the two `estimates` above, hence the variable name ;p.

And, at a quick eyeballing, `time_to_read` itself should be sane.